financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/util/ServiceExtensionUtils.java

```diff
@@ -78,28 +81,55 @@ public static JSONObject invokeExternalServiceCall(ExternalServiceRequest extern
             httpPost.setEntity(params);
             httpPost.setHeader(FinancialServicesConstants.CONTENT_TYPE_TAG,
                     FinancialServicesConstants.JSON_CONTENT_TYPE);
-            httpPost.setHeader(FinancialServicesConstants.ACCEPT, FinancialServicesConstants.JSON_CONTENT_TYPE);
+            httpPost.setHeader(FinancialServicesConstants.ACCEPT,
+                    FinancialServicesConstants.JSON_CONTENT_TYPE);
 
             // TODO: need to see security requirement
 
             CloseableHttpResponse response = HTTPClientUtils.getHttpClient().execute(httpPost);
+            HttpEntity entity = response.getEntity();
+            if (entity == null) {
+                throw new FinancialServicesException("Error occurred while invoking the external service");
+            }
+
+            InputStream inputStream = entity.getContent();
+            if (inputStream == null) {
+                throw new FinancialServicesException("Error occurred while invoking the external service");
+            }
 
-            if (response.getStatusLine().getStatusCode() != 200) {
-                String errorMsg = IOUtils.toString(response.getEntity().getContent());
-                if (log.isDebugEnabled()) {
-                    log.debug(String.format("Error occurred while invoking the external service. " +
-                                    "Status code: %s, Error: %s", response.getStatusLine().getStatusCode(),
-                            errorMsg.replaceAll("[\r\n]", "")));
-                }
-                throw new FinancialServicesException("Error occurred while invoking the external service: " + errorMsg);
+            String responseContent = IOUtils.toString(inputStream, String.valueOf(StandardCharsets.UTF_8));
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode != 200) {
+                log.error(String.format("Error occurred while invoking the external service. " +
+                                "Status code: %s, Error: %s", statusCode,
+                        responseContent.replaceAll("[\r\n]", "")));
+                throw new FinancialServicesException("Error occurred while invoking the external service");
             }
-            InputStream in = response.getEntity().getContent();
-            return new JSONObject(IOUtils.toString(in, String.valueOf(StandardCharsets.UTF_8)));
+
+            // Map response to response model
+            return mapResponse(responseContent, ExternalServiceResponse.class);
+        } catch (JsonProcessingException e) {
+            throw new FinancialServicesException("Error occurred while mapping response to model class", e);
         } catch (FinancialServicesException | IOException e) {
```

Comment:
Why are we catching FinancialServicesException here?

Reply:
fixed with ea4c439e02b2d06e21cc7c2bfcc6785b9b2a7c7f